### PR TITLE
Nuget .NET changes for Mac Catalyst

### PR DESCRIPTION
**Pull request by YUNQIUGUO**
### Description

Add Nuget package changes for adding new 'net6.0-maccatalyst' platform.

The output ORT Nuget package was manually tested and verified in a .NET MAUI app setup.

### CODE REVIEW

**Review by snnn — APPROVED** [review]

File: csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj [hunk 1]
```diff
@@ -39,7 +39,7 @@
   <PropertyGroup Condition="('$(OrtPackageId)' == 'Microsoft.ML.OnnxRuntime' OR
                              '$(OrtPackageId)' == 'Microsoft.ML.OnnxRuntime.Azure') AND
                             '$(IncludeMobileTargets)' == 'true'">
-    <MobileTargets>$(MobileTargets);net6.0-android;net6.0-ios</MobileTargets>
+    <MobileTargets>$(MobileTargets);net6.0-android;net6.0-ios;net6.0-maccatalyst</MobileTargets>
```
Comment (skottmckay) [inline csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj:42]: net6 is EOL. can we change all the net6 values in the C# csproj files to net7?

File: csharp/src/Microsoft.ML.OnnxRuntime/targets/net6.0-maccatalyst/_._
Comment (skottmckay) [inline csharp/src/Microsoft.ML.OnnxRuntime/targets/net6.0-maccatalyst/_._:1]: Should we have a readme in this directory or the parent directory explaining the usage of the '_._' empty file?

Comment (YUNQIUGUO) [reply csharp/src/Microsoft.ML.OnnxRuntime/targets/net6.0-maccatalyst/_._:1]: cool - added a brief README.md for explaining.

File: csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj (hunk 1 above)
Comment (YUNQIUGUO) [reply csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj]: yes, updated all the net6 -> 7. let me know if any other places missing or places need to be kept where I overlooked.

File: tools/ci_build/github/azure-pipelines/templates/jobs/win-ci-vs-2022-job.yml [hunk 2]
```diff
@@ -96,6 +96,18 @@ jobs:
...
+    - task: UseDotNet@2
+      inputs:
+        version: '7.x'
+      env:
+        PROCESSOR_ARCHITECTURE: 'x86'
+
+  - ${{ if contains(parameters.additionalBuildFlags, 'use_cuda') }}:
+    - task: UseDotNet@2
+      inputs:
+        version: '7.x'
```
Comment (skottmckay) [inline tools/ci_build/github/azure-pipelines/templates/jobs/win-ci-vs-2022-job.yml:109]: Would it be simpler to instead always have a UseDotNet@2 task, with an 'if' to set processor architecture, and an 'else' to not?

**Review by skottmckay — APPROVED** [review]

**Review by skottmckay — APPROVED** [review]

**Pull request merged**